src/Base/Model/Model.php

```diff
@@ -19,24 +20,62 @@ class Model implements StorableArrayable, ArrayAccess, ModelInterface
 {
     use HasAttributes;
 
+    /**
+     * @var array
+     */
+    public static $booted;
+
+    /**
+     * @var array|mixed
+     */
+    protected static $traitInitializers;
+
     /**
      * @var bool
      */
     protected $cloned = false;
 
     /**
      * @param  null|array $data
+     *
+     * @throws \ReflectionException
      */
     public function __construct(?array $data = null)
     {
         $this->guarded    = Utils::changeArrayKeysCase($this->guarded);
         $this->attributes = $this->guarded + Utils::changeArrayKeysCase($this->attributes);
 
+        $this->bootIfNotBooted();
+
+        $this->initializeTraits();
+
         $convertedData = Utils::changeArrayKeysCase($data ?? []);
 
         $this->fill($convertedData + $this->attributes);
     }
 
+    /**
+     * @return void
+     * @throws \ReflectionException
+     */
+    protected static function bootTraits(): void
+    {
+        $class = static::class;
+
+        static::$traitInitializers[$class] = [];
+
+        $traits = Utils::getClassTraitsRecursive($class);
+
+        foreach ($traits as $trait) {
+            $baseName = (new ReflectionClass($trait))->getShortName();
```

Comment:
Je moet reflection in prod code vermijden ivm performance. Hier is `Utils::classBasename ` voor